### DISCUSSION TITLE
Help getting SQL Error Codes?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [aarondl/sqlboiler#1005] [Q&A] [answered]
For example, if I try to insert a row in which the primary key already exists (postgres/CRDB), I would like to be able to look at the corresponding code `23505` to determine how the handle the error.

I'm struggling to find out how to extract these error codes from the error returned.

**Comment by aarondl** [comment, marked as answer]
`errors.As` should work with this library for introspecting errors that are returned.
